plugin/keyring_vault/vault_base64.cc

```diff
@@ -0,0 +1,57 @@
+#include <algorithm>
+#include "vault_base64.h"
+#include "base64.h"
+#include "boost/scoped_array.hpp"
+#include "boost/move/unique_ptr.hpp"
+
+namespace keyring
+{
+  bool Vault_base64::encode(const void *src, size_t src_len, Secure_string *encoded, Base64Format format)
+  {
+    uint64 memory_needed = base64_needed_encoded_length(src_len);
+    boost::scoped_array<char> base64_encoded_text(new char[memory_needed]);
```

Comment:
I need to pass array pointer to function ::base64_encoded, which will in turn fill this array with encoded text. I cannot use:
- vector - because I cannot access directly memory stored in vector, thus I cannot get pointer to this memory. I can use begin(), but this returns iterator, which I need to cast to a pointer, or &vector[0], but for this to work vector cannot be empty and calling reserve beforehand does not help. 
- string - the same logic as for string